django/contrib/admin/templates/admin/edit_inline/tabular.html

```diff
@@ -1,66 +1,102 @@
-             id="{{ inline_admin_formset.formset.prefix }}-{% if forloop.last and inline_admin_formset.has_add_permission %}empty{% else %}{{ forloop.counter0 }}{% endif %}">
-        <td class="original">
-          {% if inline_admin_form.original or inline_admin_form.show_url %}<p>
-          {% if inline_admin_form.original %}
-          {{ inline_admin_form.original }}
-          {% if inline_admin_form.model_admin.show_change_link and inline_admin_form.model_admin.has_registered_model %}<a href="{% url inline_admin_form.model_admin.opts|admin_urlname:'change' inline_admin_form.original.pk|admin_urlquote %}" class="{{ inline_admin_formset.has_change_permission|yesno:'inlinechangelink,inlineviewlink' }}">{% if inline_admin_formset.has_change_permission %}{% translate "Change" %}{% else %}{% translate "View" %}{% endif %}</a>{% endif %}
-          {% endif %}
-          {% if inline_admin_form.show_url %}<a href="{{ inline_admin_form.absolute_url }}">{% translate "View on site" %}</a>{% endif %}
-            </p>{% endif %}
-          {% if inline_admin_form.needs_explicit_pk_field %}{{ inline_admin_form.pk_field.field }}{% endif %}
-          {% if inline_admin_form.fk_field %}{{ inline_admin_form.fk_field.field }}{% endif %}
-        </td>
-        {% for fieldset in inline_admin_form %}
-          {% for line in fieldset %}
-            {% for field in line %}
-              <td class="{% if field.field.name %}field-{{ field.field.name }}{% endif %}{% if field.field.is_hidden %} hidden{% endif %}">
-              {% if field.is_readonly %}
-                  <p>{{ field.contents }}</p>
-              {% else %}
-                  {{ field.field.errors.as_ul }}
-                  {{ field.field }}
+        {{ inline_admin_formset.formset.non_form_errors }}
+      {% endblock %}
+      <table>
+        <thead>
+          <tr>
+            <th class="original"></th>
+            {% block table-headers %}
+              {% for field in inline_admin_formset.fields %}
+                <th class="column-{{ field.name }}{% if field.required %} required{% endif %}{% if field.widget.is_hidden %} hidden{% endif %}">{{ field.label|capfirst }}
+                  {% if field.help_text %}
+                    <img src="{% static "admin/img/icon-unknown.svg" %}" class="help help-tooltip" width="10" height="10" alt="({{ field.help_text|striptags }})" title="{{ field.help_text|striptags }}">
+                  {% endif %}
+                </th>
+              {% endfor %}
+            {% endblock %}
+            {% block table-delete-header %}
+              {% if inline_admin_formset.formset.can_delete and inline_admin_formset.has_delete_permission %}
+                <th>
+                  {% translate "Delete?" %}
+                </th>
+              {% endif %}
+            {% endblock %}
+          </tr>
+        </thead>
+
+        <tbody>
+          {% block table-body %}
```

Comment:
This block is problematic, splitting it into blocks would be a bit inconvenient. With this split I think it would be mostly used as block.super and then extending it. I'll ask in the forum.